### ISSUE TITLE
Make find() constexpr in C++11

### ISSUE THREAD

**Issue by eyalroz** [martinmoene/string-view-lite#48]
Currently, the `find()` method of basic_string_view is nssv_constexpr14. It seems to me it should be possible to make it constexpr since C++11 with tail-recursion iteration.

**Comment by eyalroz** [comment]
@martinmoene : Is the naive implementation not ok for the constexpr case?

**Comment by eyalroz** [comment]
@martinmoene : See #49 .

**Comment by martinmoene** [comment]
@eyalroz thanks for your input! It helps to get me going :)

**Comment by eyalroz** [comment]
@martinmoene : Do you think we could possibly use an immediately-invoked lambda instead of the function call?

**Comment by eyalroz** [comment]
Right, we wouldn't be able to recurse. I just wanted to avoid the separate function outside the class, if it's only used in one place.
